services/wallet/activity/activity.go

```diff
@@ -734,9 +742,9 @@ func getActivityEntries(ctx context.Context, deps FilterDependencies, addresses
 			// Extract tokens and chains
 			var involvedToken *Token
 			if tokenAddress != nil && *tokenAddress != ZeroAddress {
-				involvedToken = &Token{TokenType: Erc20, ChainID: common.ChainID(chainID.Int64), Address: *tokenAddress}
+				involvedToken = &Token{TokenType: Erc20, ChainID: common.ChainID(chainID.Int64), TokenID: tokenID, Address: *tokenAddress}
```

Comment:
tokenID is exclusive to ERC721 (and ERC1155), shouldn't we be passing a nil here?